internal/test/framework/windowsvm.go

```diff
@@ -146,6 +149,63 @@ func (w *windowsVM) CopyFile(filePath, remoteDir string) error {
 	return nil
 }
 
+// RetrieveFiles retrieves list of files from remote directory to the local directory.
+// As of now, I think there is no need to have a recursive copy as the log directory should contain only log files
+//but no child directories. I am fine change the implementation if the use-case arises.
+func (w *windowsVM) RetrieveFiles(remoteDir, localDir string) error {
+	// Create local dir
+	err := os.MkdirAll(localDir, os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("could not create %s: %s", localDir, err)
+	}
+
+	if w.sshClient == nil {
+		return fmt.Errorf("RetrieveFile cannot be called without a ssh client")
+	}
+
+	sftp, err := sftp.NewClient(w.sshClient)
+	if err != nil {
+		return fmt.Errorf("sftp initialization failed: %v", err)
+	}
+	defer sftp.Close()
+
+	// Get the list of all files in the directory
+	remoteFiles, err := sftp.ReadDir(remoteDir)
+	if err != nil {
+		return fmt.Errorf("error opening remote file: %v", err)
+	}
+
+	for _, remoteFile := range remoteFiles {
+		// Assumption: We ignore the directories here the reason being RetrieveFiles should just retrieve files
+		// in a directory, if this is directory, we should have called RetrieveFiles on this directory
+		if remoteFile.IsDir() {
+			continue
+		}
+		fileName := remoteFile.Name()
+		dstFile, err := os.Create(filepath.Join(localDir, fileName))
+		if err != nil {
+			return fmt.Errorf("error creating file locally: %v", err)
+		}
+		// TODO: Check if there is some performance implication of multiple Open calls.
+		srcFile, err := sftp.Open(remoteDir + "\\" + fileName)
```

Comment:
I don't know about performance implications but you should close them after the copy is complete.

Reply:
Good point. I missed it. Thanks for catching it.

Reply:
Done